src/integrationTest/java/uk/gov/hmcts/reform/userprofileapi/integration/UpdateUserProfileIntTest.java

```diff
@@ -98,6 +98,55 @@ public void should_return_200_and_when_IdamStatus_is_updated() throws Exception
 
     }
 
+    @Test
+    public void should_see_idam_error_message_and_when_IdamStatus_is_updated_by_exui_and_idam_fails() throws Exception {
+
+
+        UserProfile persistedUserProfile = userProfileMap.get("user");
+        persistedUserProfile.setStatus(IdamStatus.ACTIVE);
+        userProfileRepository.save(persistedUserProfile);
+        String idamId = persistedUserProfile.getIdamId();
+        UpdateUserProfileData data = buildUpdateUserProfileData();
+        data.setIdamStatus("Active");
+        setSidamUserUpdateMockWithStatus(NOT_FOUND.value(), false, idamId);
+        UserProfileRolesResponse userProfileRolesResponse = userProfileRequestHandlerTest.sendPut(
+                mockMvc,
+                APP_BASE_PATH + SLASH + idamId + "?origin=EXUI",
+                data,
+                NOT_FOUND,
```

Comment:
most of the code is similar in the both test case ...apart from boolean flag..please reuse the code

Reply:
done